.github/workflows/publishVSCode.yml

```diff
@@ -0,0 +1,107 @@
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        type: string
+        required: true
+    outputs:
+      result:
+        description: Set to published if the push to the VSCode marketplace was successful
+        value: ${{ jobs.publish.outputs.result }}
+      status: 
+        description: Set to complete if all tasks assoicated with the workflow were successful.
+        value: ${{ jobs.mergeDevelop.outputs.status }}
+
+# There are two outputs that are a result of running this workflow
+# The steps.publsh.outputs.result is an indicator that the pulish to the vscode marketplace succeeded.
+# The steps.mergeDevelop.outputs.status is an indicator that all the publish steps completed.
+jobs: 
+  checkout-code: 
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: '16.13.x'
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        ref: 'main'
+        token: ${{ secrets.IDEE_GH_TOKEN }}
+    - uses: ./.github/actions/gitConfig
+      with:
+        email: ${{ secrets.IDEE_GH_EMAIL }}
+    - run: npm ci
+  updating-sha-for-publish: 
+    name: 'Update SHA256 File'
+    runs-on: ubuntu-latest
+    steps:
+      - run: npm run vscode:sha256
+      - run: scripts/concatenate-sha256.js
+      - run: rm ./SHA256
+      - run: |
+              git add SHA256.md
+              git commit -m "chore: updated SHA256 v${RELEASE_VERSION}"
+      - run: echo "SHA256.md successfully generated and committed."
+  get-artifacts:
+    name: 'Download Artifacts'
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/download-artifact@v3
+        with:
+          name: VS Code Extensions
+          path: ./extensions
+      - name: Display downloaded vsix files
+        run: ls -R ./extensions
+  publish:
+    needs: ['get-artifacts', 'updating-sha-for-publish']
+    outputs: 
+      result: ${{ steps.vsce.outputs.result }}
+    runs-on: ubuntu-latest
+    env: 
+      VSCE_PERSONAL_ACCESS_TOKEN: ${{ secrets.VSCE_PERSONAL_ACCESS_TOKEN }}
+    steps: 
+      - run: find ./extensions -type f -name "*.vsix" -exec vsce publish --pat ${{ env.VSCE_PERSONAL_ACCESS_TOKEN }} --packagePath {} \;
+      - id: vsce
+        run: | 
+          echo "result=published" >> $GITHUB_OUTPUT
+      - run: echo "SUCCESSFULLY published"
+  gitTag: 
+    needs: ['publish']
+    runs-on: ubuntu-latest
+    steps: 
+      - run: |
+          git tag v${{ inputs.version }}
+          git push origin v${{ inputs.version }}
+  pushMain: 
+    needs: [publish]
+    runs-on: ubuntu-latest
+    steps: 
+      - run: git push origin main
+  githubRelease:
+    needs: [pushMain]
+    runs-on: ubuntu-latest
+    steps: 
+      - run: gh release create v${{ inputs.version }} ./extensions/*.vsix --title "Release v${{ inputs.version }}" --notes-file ./packages/salesforcedx-vscode/CHANGELOG.md
+        env:
+          GITHUB_TOKEN: ${{ secrets.IDEE_GH_TOKEN }}
+  mergeDevelop:
```

Comment:
note these jobs all depend on the previous step completing.  My intention here is if a step fails due to a transient error we can rerun from failed and complete the publish steps without too much manual intervention.

Reply:
yeah I thought about that as well but in the end figured it'd be simpler to get all this working then adjust if it makes sense.  It does seem like it would make sense to publish on a tag creation, but I didn't want to alter the current flow until we get a better feel for it.